src/DropdownToggle.js

```diff
@@ -49,6 +47,16 @@ class DropdownToggle extends React.Component {
     this.context.toggle(e);
   }
 
+  getRole() {
+    const menuRole = this.context.menuRole
+    if(menuRole === true || menuRole === 'menu') {
```

Comment:
We can just return `menuRole` here since all the supported values in `menuRole` are also applicable to `aria-haspopup`.  So this whole function can be simplified to return `context.menuRole` if it exists, otherwise return the `aria-haspopup` prop.

Reply:
Scratch my comment above.  `menuRole` could be undefined here.

Should do something like this... 

```
return this.context.menuRole || this.props.['aria-haspopup'];
```